### DISCUSSION TITLE
Correct implementation of the "isDirty" property

### DISCUSSION THREAD

**Discussion by aheadweb** [pmndrs/zustand#2288] [Q&A] [answered]
Hi, one problem got me thinking. I'll try to describe it. I want to hear your opinion

What we have:
```typescript
interface Store {
  company?: number;
  catalog?: number;
  product:? string;
  dateUpload?: string;
}
```
_How to correctly solve the problem “Has at least 1 field been changed?”_
I came to this decision
```typescript
export const isDirtyStoreSelector = (store: Store ) =>
  store.catalog ||
  store.company ||
  store.dateTaken ||
  store.dateUpload
```
But what if there are more properties. The selector will grow larger and larger.
It seems to me that this solution is not beautiful, perhaps there is a more interesting approach.

**Comment by dbritto-dev** [comment, marked as answer]
@aheadweb creating a new field call `_isDirty` and subscribing to your store and update `_isDirty` to `true` only when is `false` (to avoid extra renders) so you can use it like this:

```ts
export const isDirtyStoreSelector = (store: Store) => store._isDirty
```

types would be something like this:

```ts
interface DirtyStore {
  _isDirty: boolean
}

interface Store extends DirtyStore {
  ...
}
```

**Reply by aheadweb** [reply]
Do you mean something like this?
```typescript
const unSub = useGalleryFilterStore.subscribe(() => {
  useGalleryFilterStore.setState({ _isDirty: true });
  unSub();
});
```

**Reply by dbritto-dev** [reply]
@aheadweb that could work
